helpers/encodeCallData.ts

```diff
@@ -0,0 +1,64 @@
+import BigNumber from "bignumber.js";
+import { numberToPaddedHex } from "@multiversx/sdk-core/out/utils.codec";
+
+const uint32ArgBytes = 8
+const uint64ArgBytes = 16
+const DataPresentProtocolMarker = "0x01"
+/**
+ * Encodes call data for the deposit function with simplified argument handling.
+ * Automatically attempts to detect and encode integers and strings.
+ *
+ * @param {string} endpointName - The name of the endpoint for the cross-chain call.
+ * @param {number} gasLimit - The gas limit for the cross-chain call.
+ * @param {any[]} args - The arguments for the cross-chain call, assumed to be strings or numbers.
+ * @returns {string} The encoded callData string.
+ */
+export function encodeCallData(endpointName: string, gasLimit: number, args: any[]): string {
+    let callData = DataPresentProtocolMarker;
+    callData = addEndpointName(callData, endpointName);
+    callData = addGasLimit(callData, gasLimit);
+    callData = addArgs(callData, args);
+
+    return callData;
+}
+
+export function addEndpointName(callData: string, endpointName: string): string {
```

Comment:
Only for internal use and tests, right (as the ones below)?

Reply:
yes